client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -1382,11 +1397,7 @@
                 };
 
                 scope.$watch('options', function() {
-                    fetchTemplates(scope.options.page, scope.options.keyword)
-                    .then(function(result) {
-                        scope.maxPage = Math.ceil(result._meta.total / PAGE_SIZE);
-                        scope.templates = result;
-                    });
+                    fetchTemplates();
                 }, true);
```

Comment:
wouldn't this work with `$watchCollection` ? imo we should avoid recursive watches